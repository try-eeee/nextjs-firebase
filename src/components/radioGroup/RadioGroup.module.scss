.radioGroup {
  display: flex;
  flex-direction: column;
}

.radioLabel {
  display: flex;
  align-items: center;
  margin: 0.5rem 0;
  padding-right: 0.85rem;

  input[type="radio"] {
    display: none;
  }

  .radioButton {
    width: 1.5rem;
    height: 1.5rem;
    border-radius: 50%;
    border: 2px solid #ccc;
    background-color: #fff;
    margin-right: 0.5rem;
    display: inline-block;
    position: relative;

    &::after {
      content: "";
      width: 0.65rem;
      height: 0.65rem;
      border-radius: 50%;
      background-color: #0070f3;
      position: absolute;
      top: 25%;
      left: 25%;
      opacity: 0;
      transition: opacity 0.2s ease;
    }
  }
}

.errorMessage {
  color: red;
  font-size: 0.875rem;
  margin-top: 0.5rem;
}

.radioLabel input[type="radio"]:checked + .radioButton::after {
  opacity: 1;
}

@media (min-width: 768px) {
  .radioGroup {
    flex-direction: row; /* PCでは横並び */
  }
}
