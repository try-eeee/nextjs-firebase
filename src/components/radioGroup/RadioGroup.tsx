import React from "react";
import styles from "./RadioGroup.module.scss";
import { UseFormRegisterReturn } from "react-hook-form";

interface RadioOption {
  value: string;
  label: string;
}

interface RadioGroupProps extends React.ComponentPropsWithoutRef<"input"> {
  options: RadioOption[];
  register?: UseFormRegisterReturn;
}

export function RadioGroup(props: RadioGroupProps) {
  const { options, register } = props;
  return (
    <div className={`${styles.radioGroup}`}>
      {options.map((option) => (
        <label key={option.value} className={styles.radioLabel}>
          <input type="radio" {...register} value={option.value} />
          <span className={styles.radioButton}></span>
          {option.label}
        </label>
      ))}
    </div>
  );
}
