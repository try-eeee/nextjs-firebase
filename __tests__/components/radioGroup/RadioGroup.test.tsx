import React from "react";
import { render, renderHook } from "@testing-library/react";
import { useForm, UseFormRegisterReturn } from "react-hook-form";
import { RadioGroup } from "@/components/radioGroup/RadioGroup";

describe("RadioGroup UnitTest", () => {
  const reactHookFormMock = renderHook(() =>
    useForm<{ gender: string }>({ defaultValues: { gender: "" } })
  );
  const { register, getValues, reset } = reactHookFormMock.result.current;

  test("Snapshot Test Of RadioGroup", () => {
    const { asFragment } = render(
      <RadioGroup
        register={register("gender")}
        options={[
          { value: "male", label: "男性" },
          { value: "female", label: "女性" },
          { value: "other", label: "その他" },
        ]}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
