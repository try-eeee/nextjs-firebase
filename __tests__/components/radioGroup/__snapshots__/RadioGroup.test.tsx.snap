// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup UnitTest Snapshot Test Of RadioGroup 1`] = `
<DocumentFragment>
  <div
    class="radioGroup"
  >
    <label
      class="radioLabel"
    >
      <input
        name="gender"
        type="radio"
        value="male"
      />
      <span
        class="radioButton"
      />
      男性
    </label>
    <label
      class="radioLabel"
    >
      <input
        name="gender"
        type="radio"
        value="female"
      />
      <span
        class="radioButton"
      />
      女性
    </label>
    <label
      class="radioLabel"
    >
      <input
        name="gender"
        type="radio"
        value="other"
      />
      <span
        class="radioButton"
      />
      その他
    </label>
  </div>
</DocumentFragment>
`;
